### ISSUE TITLE
[cm] list continuation indentation

### ISSUE THREAD

**Issue by richburdon** [dxos/dxos#7678]
## Current
- We add styles so that continuation lines are indented in lists.
- The markdown extension adds the space after the list mark to the beginning of the following text span; we subtract `0.25em` to account for this, but this causes the cursor and selections to be indented.

```html
<div class="cm-line" style="padding-left: 24px; text-indent: calc(-0.25em - 24px">
  <span class="cm-list-mark">1.</span>
  <span> Hello world</span>
</div>
```

## Question
- Can we remove the space from the text.

![Image](https://github.com/user-attachments/assets/dee6f6ac-c79a-41f9-be11-d13a39b1dd28)

**Comment by marijnh** [comment]
> I'm a little uneasy that the decoration ranges are now not aligned with the token ranges.

No that's absolutely not a problem.

I did not really manage to understand the initial source of the issue here. But if you've found a workaround maybe that's not relevant anymore anyway.

**Comment by richburdon** [comment]
A minor issue remains: when creating a new list item, the cursor is offset from the line before the next character is typed:

## Example

Before

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/34099e6c-871d-404c-92b6-6565940e64a2" />

After

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/34edcdf7-92ec-43ba-9d70-44ce1d996552" />

## Demo

https://dxos-storybook.pages.dev/?path=/story/ui-react-ui-editor-texteditor--task-list

**Comment by marijnh** [comment]
> Is there anything wrong with our styles?

No, this is because CodeMirror derives its cursor height from the content around it. This is necessary because things can be styled arbitrarily so there's no definitive editor-wide (or even line-wide) cursor size. When there's no text next to the cursor it uses whatever content is there. In this case, it'll use the size of the widget buffer after the checkbox. I think you might be able to use a [`coordsAt`](https://codemirror.net/docs/ref/#view.WidgetType.coordsAt) method on the widget to manually influence this height.
